pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -68,23 +69,22 @@ func (r *TokenREST) Create(ctx genericapirequest.Context, name string, obj runti
 		var uid types.UID
 
 		gvk := schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind)
+		var err error
 		switch {
 		case gvk.Group == "" && gvk.Kind == "Pod":
-			podObj, err := r.pods.Get(ctx, ref.Name, &metav1.GetOptions{})
+			pod, err = r.pods.Pods(svcacct.Namespace).Get(ref.Name, metav1.GetOptions{})
```

Comment:
This will actually do an http rpc. I think you want to just override RequestInfo.

Reply:
or create a new context with the same namespace as ctx

Reply:
Done.
Thanks for your suggestions.